autorest/authorization.go

```diff
@@ -44,6 +47,53 @@ func (na NullAuthorizer) WithAuthorization() PrepareDecorator {
 	return WithNothing()
 }
 
+// APIKeyAuthorizer implements API Key authorization.
+type APIKeyAuthorizer struct {
```

Comment:
I am not a big fan of Azure specific code to live in the `autorest` folder

Reply:
the whole code in the repo is in the `autorest` folder. If we want to change that lets make an issue and discuss it since we need to move other stuff out like `msi` also